### PR TITLE
chore(broker-core): correlate message to event-based gateway

### LINKED ISSUE #1709: Support message catch events attached to an event-based gateway
When correlating a workflow instance subscription, we currently check if the associated element is active in the `ElementInstanceState`; since event based gateways are not elements, the correlate command is rejected.

Everything else seems to work apart from the correlation, as far as I can tell.

Additionally, should validate that we aren't expecting the same message for two different catch elements attached to the same gateway.

### PR DESCRIPTION
**Pull request by saig0**
closes #1709

### CODE REVIEW

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/EventBasedGatewayValidator.java [hunk 1]
```diff
@@ -82,4 +96,16 @@ private boolean isValidEvent(final IntermediateCatchEvent event) {
           .anyMatch(e -> e.isAssignableFrom(eventDefinition.getClass()));
     }
   }
+
+  private Stream<MessageEventDefinition> getMessageEventDefinitions(
+      Collection<SequenceFlow> outgoingSequenceFlows) {
+    return outgoingSequenceFlows
+        .stream()
+        .map(SequenceFlow::getTarget)
+        .filter(t -> t instanceof IntermediateCatchEvent)
```
Comment (npepinpe) [inline bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/EventBasedGatewayValidator.java]: Wouldn't an event based gateway only have outgoing flows to `IntermediateCatchEvent`?

**Review by npepinpe — APPROVED** [review]
Looks good :+1:

**Comment by menski** [comment]
bors retry

**Comment by menski** [comment]
This branch fails after merge with develop in the model api validation tests:

```
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.001 s <<< FAILURE! - in io.zeebe.model.bpmn.validation.ZeebeEventBasedGatewayValidationTest

[ERROR] validateModel[1: [ERROR: {type: IntermediateCatchEvent, message: Must have a time duration}]](io.zeebe.model.bpmn.validation.ZeebeEventBasedGatewayValidationTest)  Time elapsed: 0.001 s  <<< FAILURE!

java.lang.AssertionError: 

Not all expecations were matched by results (or vice versa)



Unmatched expectations:

ERROR: {type: IntermediateCatchEvent, message: Must have a time duration}



Unmatched results:

ERROR: {id: timerEventDefinition_390001c0-c1dd-4bcd-9d12-bdf602212a3f, type: TimerEventDefinition, message: Time cycles must be non-interrupting boundary events}
```

**Pull request merged**